core/rawdb/chain_freezer.go

```diff
@@ -276,6 +277,11 @@ func (f *chainFreezer) freezeRange(nfdb *nofreezedb, number, limit uint64) (hash
 			if len(td) == 0 {
 				return fmt.Errorf("total difficulty missing, can't freeze block %d", number)
 			}
+			// TODO(GalaIO): blobs may nil before cancun fork
+			blobs := ReadBlobsRLP(nfdb, hash, number)
+			//if len(blobs) == 0 {
```

Comment:
Can't we check if it is cancun fork or not and have this check? Or will it be part of your TODO mentioned above?